projects/packages/search/src/widgets/class-search-widget.php

```diff
@@ -324,12 +324,17 @@ public function widget_non_instant( $args, $instance ) {
 		$display_filters = false;
 
 		// Search instance must have been initialized before widget render.
-		if ( is_search() && Classic_Search::instance() ) {
+		if ( is_search() ) {
+			if ( Inline_Search::should_replace_classic_search() ) {
```

Comment:
I am not sure if it is feasible, but we could move this block to a function in the `class-initializer.php` since it is used there and here for the widget.

Reply:
Makes sense, I've introduced `get_instance_maybe_fallback_to_classic` method to address this.